pkg/local_object_storage/engine/inhume.go

```diff
@@ -227,6 +234,23 @@ func (e *StorageEngine) inhumeAddr(addr oid.Address, prm shard.InhumePrm) (bool,
 
 	prm.SetTargets(append(children, addr)...)
 
+	if shardWithObject != "" {
+		sh := e.getShard(shardWithObject)
+
+		_, err := sh.Inhume(prm)
+		if err != nil {
```

Comment:
How come error handling differs from the one below here?

Reply:
here we know what exact shard has an object, no need to return `stop`, `retErr`, just `return`

if you are about why we need to compare error with smth and return that error in the code below -- i don't know, all of the cases below are logical errors anyway

Reply:
Yeah, I'm mostly about ErrLockObjectRemoval and ErrReadOnlyMode/ErrDegradedMode.

Reply:
iteration over shards can be stopped if some logical errors appear (lock(ed) object removal, degraded mode, etc), so that is why we look at some kinds of errors. if we have exact one shard, we can return `err` directly

(but this meta -> shard -> meta conversion scares me too: 
```
			case errors.Is(err, shard.ErrLockObjectRemoval):
				retErr = meta.ErrLockObjectRemoval
				return true
```
, but that is not used, and `errors.Is` will work the same)